spec/models/manageiq/providers/redhat/infra_manager/provision_workflow_spec.rb

```diff
@@ -128,6 +128,21 @@ def allowed_customization_templates(options)
       expect(workflow).to receive(:super_allowed_customization_templates).with(options)
       workflow.allowed_customization_templates(options)
     end
+
+    it "should retrieve templates in region" do
+      FactoryGirl.create(:customization_template_cloud_init, :name => "test")
+
+      my_region_number = template.my_region_number
+      other_region_id  = (my_region_number + 1) * template.class.rails_sequence_factor + 1
+      pxe_image_type   = FactoryGirl.create(:pxe_image_type, :name => "test_image", :id => other_region_id)
+      FactoryGirl.create(:customization_template_cloud_init,
+                         :name           => "test",
+                         :id             => other_region_id,
+                         :pxe_image_type => pxe_image_type)
+
+      allow(workflow).to receive(:supports_native_clone?).and_return(true)
+      expect(workflow.allowed_customization_templates.size).to eq(1)
```

Comment:
I think it would be better for this to `match_array` to verify that we end up with the correct template.